### ISSUE TITLE
Incompatible pickle cache support on Python 2.7

### ISSUE THREAD

**Issue by karan6181** [horovod/horovod#1681]
**Environment:**
1. Framework: MXNet
2. Framework version: 1.5.1
3. Horovod version: 0.19.0
4. MPI version: 3.1.0
5. CUDA version: 10.0
6. NCCL version: 2.4.8
7. Python version: 2.7
8. OS and version: Ubuntu 16.04
9. GCC version: 5.4.0

I was trying to install and run horovod with MXNet backend on a linux machine(EC2_GPU) with python 2.7. I was able to install horovod successfully but unable to run some example script(e.g. mxnet_mnist.py). I saw below error:

```
There is an error with reading cache file. You can delete the corrupt file: /home/ubuntu/.horovod/cache.bin.
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/mxnet_p27/bin/horovodrun", line 21, in <module>
    run_commandline()
  File "/home/ubuntu/anaconda3/envs/mxnet_p27/lib/python2.7/site-packages/horovod/run/run.py", line 867, in run_commandline
    _run(args)
  File "/home/ubuntu/anaconda3/envs/mxnet_p27/lib/python2.7/site-packages/horovod/run/run.py", line 756, in _run
    parameters_hash)
  File "/home/ubuntu/anaconda3/envs/mxnet_p27/lib/python2.7/site-packages/horovod/run/util/cache.py", line 45, in __init__
    content = cloudpickle.load(cf)
  File "/home/ubuntu/anaconda3/envs/mxnet_p27/lib/python2.7/pickle.py", line 1384, in load
    return Unpickler(file).load()
  File "/home/ubuntu/anaconda3/envs/mxnet_p27/lib/python2.7/pickle.py", line 864, in load
    dispatch[key](self)
  File "/home/ubuntu/anaconda3/envs/mxnet_p27/lib/python2.7/pickle.py", line 892, in load_proto
    raise ValueError, "unsupported pickle protocol: %d" % proto
ValueError: unsupported pickle protocol: 4
```

**Comment by tgaddair** [comment]
@abditag2, can you take a look?  Basically, a cache created with Python 3.6 cannot be loaded by a Python 2.7 interpreter.  We should handle this gracefully by evicting / ignoring the cache in this instance.

**Comment by abditag2** [comment]
This should fix it: https://github.com/horovod/horovod/pull/1682

**Comment by tgaddair** [comment]
Fixed by #1682.
